cro-clib/src/types.rs

```diff
@@ -46,7 +64,62 @@ impl CroResult {
     }
 }
 
+/// copy from c array to rust array
+/// # Safety
+pub unsafe fn copy_string(src: &str, dst: &mut [u8]) {
+    dst[..src.len()].copy_from_slice(&src.as_bytes()[..src.len()]);
+    dst[src.len()] = 0;
+}
+
 /// # Safety
 pub unsafe fn get_string(src: *const c_char) -> String {
     CStr::from_ptr(src).to_string_lossy().into_owned()
 }
+
+/// # Safety
+pub unsafe fn get_string_from_array(src: &[u8]) -> String {
+    let mut n = 0;
+    for (i, item) in src.iter().enumerate() {
+        if 0 == (*item) as u8 {
+            n = i;
+            break;
+        }
+    }
+
+    std::str::from_utf8(&src[0..n]).expect("utf8").to_string()
+}
+
+#[repr(C)]
+pub struct CroUtxo {
+    pub address: [u8; 100], // holder
+    pub coin: [u8; 100],    // amount in carson unit
```

Comment:
address should be 32 bytes and amount is u64? what about time locking?

Reply:
i'll add time locking

Reply:
added timelock